link/src/main/java/com/stripe/android/link/ui/verification/VerificationViewModel.kt

```diff
@@ -0,0 +1,220 @@
+package com.stripe.android.link.ui.verification
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import androidx.lifecycle.viewmodel.initializer
+import androidx.lifecycle.viewmodel.viewModelFactory
+import com.stripe.android.core.Logger
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.link.LinkActivityResult
+import com.stripe.android.link.LinkScreen
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.analytics.LinkEventsReporter
+import com.stripe.android.link.injection.NativeLinkComponent
+import com.stripe.android.link.model.AccountStatus
+import com.stripe.android.link.ui.ErrorMessage
+import com.stripe.android.link.ui.getErrorMessage
+import com.stripe.android.ui.core.elements.OTPSpec
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/**
+ * ViewModel that handles user verification confirmation logic.
+ */
+internal class VerificationViewModel @Inject constructor(
+    private val linkAccountManager: LinkAccountManager,
+    private val linkEventsReporter: LinkEventsReporter,
+    private val logger: Logger,
+    private val goBack: () -> Unit,
+    private val navigateAndClearStack: (route: LinkScreen) -> Unit,
+    private val dismissWithResult: (LinkActivityResult) -> Unit
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(
+        value = run {
+            val linkAccount = linkAccountManager.linkAccount.value
+            if (linkAccount == null) {
+                dismissWithResult(LinkActivityResult.Failed(NoLinkAccountFoundForVerification()))
+            }
```

Comment:
There should be a link account (unverified) in the link account manager at this point